pkg/edit/config_api.go

```diff
@@ -69,6 +69,44 @@ func callHooks(ev *eval.Evaler, name string, hook vals.List, args ...interface{}
 	}
 }
 
+func callFilters(ev *eval.Evaler, name string, filters vals.List, args ...interface{}) bool {
+	i := -1
+	for it := filters.Iterator(); it.HasElem(); it.Next() {
+		i++
+		name := fmt.Sprintf("%s[%d]", name, i)
+		fn, ok := it.Elem().(eval.Callable)
+		if !ok {
+			// TODO(xiaq): This is not testable as it depends on stderr.
+			// Make it testable.
+			diag.Complainf("%s not function", name)
+			continue
+		}
+		// TODO(xiaq): This should use stdPorts, but stdPorts is currently
+		// unexported from eval.
+		ports := []*eval.Port{
+			{File: os.Stdin}, {File: os.Stdout}, {File: os.Stderr}}
+		fm := eval.NewTopFrame(ev, eval.NewInternalGoSource(name), ports)
+		out, err := fm.CaptureOutput(fn, args, eval.NoOpts)
+		if err != nil {
+			diag.Complainf("%s return error", name)
+			continue
+		}
+		if len(out) != 1 {
```

Comment:
I don't see how this enforces the requirement that the filter output a single `$true` or `$false`. And, in fact, your single test case outputs `$false`. What happens if the filter outputs `argle`, `1`, `(float64 3.14)`?

Reply:
Line 98 checks the type of the output. There should be a test for this but this is not testable at the moment as the warning message can only be written to stderr.

I plan to refactor `Evaler` a bit, including adding a `ports` field to it (now only `Frame` has ports), which will make this testable in future.